clang/test/SemaSYCL/sycl-2017-future-compat.cpp

```diff
@@ -0,0 +1,13 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -sycl-std=2017 -verify %s
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -sycl-std=2017 -Wno-sycl-2017-compat -verify=sycl-2017 %s
+
+// We do not expect any diagnostics from this file when disabling future compat
+// warnings.
+// sycl-2017-no-diagnostics
+
+// Test that we get compatibility warnings when using a SYCL 2020 attribute
+// spelling while in SYCL 2020 mode.
```

Comment:
missing 'not'

Reply:
Hehe, c'mon, maybe the test cases are what's wrong. Maybe we WANT diagnostics when using SYCL 2020 features in SYCL 2020 mode. Like "Congrats, you're using SYCL 2020. You win... uh... something, I guess." ;-)